File: scratch/enrich_products.js
const fs = require('fs');

const modelData = {
    'AQR-D185': {
        shortDesc: 'Kulkas 1 Pintu dengan Giant Freezer dan Tempered Glass Tray.',
        description: 'Aqua AQR-D185 adalah kulkas 1 pintu yang dirancang untuk efisiensi energi dan kemudahan penggunaan. Dilengkapi dengan Giant Freezer berkapasitas 23L dan rak kaca tempered yang mampu menahan beban berat.',
        specs: {
            'Kapasitas': '145 Liter',
            'Daya': '67 Watt',
            'Dimensi': '525 x 524 x 1060 mm',
            'Sistem Pendingin': 'Direct Cooling',
            'Fitur': 'Semi-Auto Defrost, R600a Ramah Lingkungan'
        },
        highlights: ['Giant Freezer 23L', 'Tempered Glass Tray tahan 100kg', 'Konsumsi daya rendah 67W'],
        sellingPoints: ['Hemat listrik bulanan', 'Garansi kompresor 7 tahun', 'Desain elegan Aurora Series'],
        objections: ['"Gampang retak raknya?" -> Tidak, sudah tempered glass tahan 100kg.', '"Berisik?" -> Sangat senyap, cocok untuk kamar atau ruang tamu.']
    },
    'P7000N': {
        shortDesc: 'Mesin Cuci 2 Tabung 7kg dengan Roller Jet Pulsator.',
        description: 'LG P7000N menawarkan performa pencucian yang kuat namun lembut pada pakaian berkat Roller Jet Pulsator. Dilengkapi dengan Wind Jet Dry untuk pengeringan yang lebih cepat.',
        specs: {
            'Kapasitas': '7.0 kg',
            'Daya Cuci': '320 Watt',
            'Dimensi': '805 x 475 x 975 mm',
            'Tipe': 'Twin Tub',
            'Fitur': 'Roller Jet Pulsator, Wind Jet Dry, 3 Wash Program'
        },
        highlights: ['Roller Jet Pulsator untuk hasil bersih maksimal', 'Wind Jet Dry mempercepat pengeringan', 'Body anti karat'],
        sellingPoints: ['Sangat awet, investasi jangka panjang', 'Hasil cucian lebih kering dibanding kompetitor', 'Mudah dioperasikan'],
        objections: ['"Ribet manual?" -> Malah lebih awet karena mekanik sederhana.', '"Rusak?" -> Garansi resmi LG terjamin di seluruh Indonesia.']
    },
    'P8000N': {
        shortDesc: 'Mesin Cuci 2 Tabung 8kg dengan Roller Jet Pulsator.',
        description: 'LG P8000N memberikan efisiensi pencucian tinggi dengan teknologi Roller Jet Pulsator. Pakaian lebih bersih dan kering lebih cepat berkat fitur Wind Jet Dry.',
        specs: {
            'Kapasitas': '8 kg',
            'Daya Cuci': '320 Watt',
            'Daya Peras': '250 Watt',
            'Dimensi': '805 x 478 x 975 mm',
            'Fitur': 'Roller Jet Pulsator, Punch+3, Wind Jet Dry, Anti Rat Cover'
        }
    },
    'P9050RTB': {
        shortDesc: 'Mesin Cuci 2 Tabung 9kg, Kuat dan Tahan Lama.',
        description: 'LG P9050RTB dirancang untuk menangani beban cucian besar hingga 9kg. Dilengkapi dengan teknologi anti tikus dan pulsator yang kuat untuk hasil cuci maksimal.',
        specs: {
            'Kapasitas': '9 kg',
            'Daya': '320 Watt (Cuci)',
            'Dimensi': '805 x 475 x 975 mm',
            'Fitur': 'Roller Jet Pulsator, Punch+3, Wind Jet Dry, Anti Rat Base'
        }
    },
    'PWM-1076': {
        shortDesc: 'Mesin Cuci Polytron 10kg dengan Hijab Mode.',
        description: 'Polytron PWM-1076 hadir dengan fitur khusus Hijab Mode untuk menjaga serat kain halus. Kapasitas 10kg sangat cocok untuk keluarga besar. Hemat energi dengan teknologi Magic Gear.',
        specs: {
            'Kapasitas': '10 kg',
            'Daya Cuci': '250 Watt',
            'Daya Peras': '130 Watt',
            'Fitur': 'Hijab Mode, Magic Gear, Big Pulsator, Water Selector'
        },
        highlights: ['Fitur khusus Hijab Mode', 'Teknologi Magic Gear hemat listrik 30%', 'Kapasitas ekstra besar 10kg'],
        sellingPoints: ['Aman untuk kain halus/hijab', 'Motor kuat & bergaransi', 'Anti karat & tahan lama'],
        objections: ['"Boros listrik?" -> Sangat hemat, sudah pakai Magic Gear.', '"Susah bersihkan filter?" -> Filter dirancang mudah dilepas-pasang.']
    },
    'ES-T70MW': {
        shortDesc: 'Mesin Cuci Sharp 6.5kg Low Watt.',
        description: 'Sharp ES-T70MW adalah mesin cuci 2 tabung yang sangat hemat listrik. Menggunakan teknologi Super Soakmagic untuk hasil rendaman yang lebih bersih.',
        specs: {
            'Kapasitas': '6.5 kg',
            'Daya Cuci': '199 Watt',
            'Fitur': 'Super Soakmagic, Silvermagic Protection, Planet Gear'
        }
    },
    'Agate-200': {
        shortDesc: 'Showcase Cooler RSA 192 Liter, Pendinginan Cepat.',
        description: 'RSA Agate-200 sangat cocok untuk mendisplay minuman dingin di toko atau restoran Anda. Menggunakan kaca anti embun dan lampu LED yang terang.',
        specs: {
            'Volume': '192 Liter',
            'Daya': '180 Watt',
            'Rak': '3 Buah',
            'Fitur': 'Fan Cooling System, LED Light, Kaca Anti Embun, Key Lock'
        }
    },
    'PAS-10DF22': {
        shortDesc: 'Speaker Aktif Polytron Double 10 inci, Bass Menggelegar.',
        description: 'Nikmati kualitas suara premium dengan Polytron PAS-10DF22. Dilengkapi Double Woofer 10 inci dan fitur karaoke lengkap untuk hiburan keluarga Anda.',
        specs: {
            'Power': '200 Watt RMS',
            'Woofer': 'Double 10 inch',
            'Koneksi': 'Bluetooth, USB, SD Card, Aux, Line In',
            'Fitur': 'Super Bass, 3-Way System, 2 Mic Input, Polytron Audio Connect'
        },
        highlights: ['Double Woofer 10 inch untuk bass kuat', 'Fitur Karaoke dengan 2 Mic Input', 'Koneksi Bluetooth canggih'],
        sellingPoints: ['Suara paling mantap di kelasnya', 'Bisa kontrol lewat smartphone', 'Desain elegan dengan LED display'],
        objections: ['"Terlalu besar?" -> Suara sebanding dengan ukurannya yang kokoh.', '"Suara pecah?" -> Tidak, sudah pakai teknologi 3-Way System.']
    },
    '43E6K': {
        shortDesc: 'Smart TV Hisense 43 inci 4K UHD Google TV.',
        description: 'Hisense 43E6K menghadirkan gambar 4K yang sangat detail dan sistem Google TV yang cerdas. Mendukung Dolby Vision dan HDR10 untuk pengalaman sinematik di rumah.',
        specs: {
            'Resolusi': '4K UHD (3840x2160)',
            'OS': 'Google TV',
            'Fitur': 'Dolby Vision, Game Mode Plus, Voice Control, HDR10'
        }
    },
    'A36 5G': {
        shortDesc: 'Samsung Galaxy A36 5G, Performa Kencang Snapdragon.',
        description: 'Galaxy A36 5G menawarkan keseimbangan sempurna antara harga dan performa. Dilengkapi layar Super AMOLED 120Hz dan kamera 50MP OIS yang handal.',
        specs: {
            'Layar': '6.7 inch Super AMOLED 120Hz',
            'Processor': 'Snapdragon 6 Gen 3',
            'RAM/Storage': '8/256 GB',
            'Kamera': '50MP OIS + 8MP + 5MP',
            'Baterai': '5000mAh, 45W Fast Charging'
        }
    },
    'A17': {
        shortDesc: 'Samsung Galaxy A17, Smartphone 5G Terjangkau.',
        description: 'Samsung Galaxy A17 hadir dengan layar luas dan baterai besar yang tahan lama. Pilihan terbaik untuk produktivitas harian dan media sosial.',
        specs: {
            'RAM/Storage': '8/256 GB',
            'Fitur': 'Layar Besar, Baterai 5000mAh, Desain Modern'
        }
    },
    'D65': {
        shortDesc: 'Sepeda Listrik U-Winfly D65, Jarak Tempuh Hingga 50km.',
        description: 'U-Winfly D65 adalah sepeda listrik yang nyaman dengan suspensi empuk. Sangat cocok untuk belanja atau mengantar anak sekolah.',
        specs: {
            'Motor': '600 Watt',
            'Baterai': '48V / 12Ah',
            'Jarak': '50 km',
            'Beban': '150 kg'
        }
    },
    'Goda 150A': {
        shortDesc: 'Sepeda Listrik Goda Golden Panda 150A.',
        description: 'Goda 150A memiliki desain yang ikonik dan motor yang bertenaga. Dilengkapi fitur keamanan lengkap untuk kenyamanan berkendara Anda.',
        specs: {
            'Motor': '550 Watt',
            'Baterai': '48V / 12Ah',
            'Jarak': '40 km',
            'Beban': '150 kg'
        }
    },
    'Goda 122': {
        shortDesc: 'Sepeda Listrik Goda 122 New Model.',
        description: 'Varian terbaru dari Goda dengan peningkatan pada daya tahan baterai dan kenyamanan tempat duduk. Cocok untuk penggunaan harian di perumahan.',
        specs: {
            'Motor': '500 Watt',
            'Baterai': '48V / 12Ah',
            'Jarak': '40 km'
        }
    },
    'Nuv F1': {
        shortDesc: 'Sepeda Listrik Nuv F1 Spiva yang Tangguh.',
        description: 'Nuv F1 Spiva dirancang dengan frame yang kuat dan ban tubeless. Memberikan stabilitas tinggi saat dikendarai di berbagai medan jalan.',
        specs: {
            'Motor': '650 Watt',
            'Baterai': '48V / 12Ah',
            'Kecepatan': '40 km/jam'
        }
    },
    'LUNA': {
        shortDesc: 'Sepeda Listrik Saige Luna Premium.',
        description: 'Saige Luna adalah kasta tertinggi dari sepeda listrik Saige. Desain sangat elegan layaknya motor matic dengan fitur serba digital.',
        specs: {
            'Motor': '800 Watt',
            'Baterai': '48V / 20Ah',
            'Jarak': '60 km',
            'Fitur': 'NFC Lock, Digital Panel, Hydraulic Brake'
        }
    },
    'RETRO': {
        shortDesc: 'Sepeda Listrik Selis Retro, Gaya Klasik Masa Kini.',
        description: 'Tampil beda dengan Selis Retro. Menggabungkan desain klasik yang timeless dengan teknologi motor listrik modern yang senyap dan bertenaga.',
        specs: {
            'Motor': '500 Watt',
            'Baterai': '48V / 20Ah',
            'Jarak': '45 km'
        }
    },
    'G-L18': {
        shortDesc: 'Sepeda Listrik Soloz G-L18, Praktis dan Modern.',
        description: 'Soloz G-L18 menawarkan kemudahan mobilitas dengan desain yang compact namun tetap bertenaga. Sangat hemat biaya operasional.',
        specs: {
            'Motor': '500 Watt',
            'Baterai': '48V / 12Ah',
            'Jarak': '40 km'
        }
    },
    'CB-171-AP': {
        shortDesc: 'Blender Cosmos 2 in 1 dengan Pisau Stainless Steel.',
        description: 'Blender Cosmos CB-171-AP sangat handal untuk menghaluskan bumbu maupun buah. Dilengkapi dengan dry mill untuk menghaluskan biji-bijian.',
        specs: {
            'Kapasitas': '1 Liter',
            'Daya': '200-380 Watt',
            'Bahan': 'Plastik Food Grade'
        }
    },
    'RI-522 CE': {
        shortDesc: 'Kompor Gas Rinnai 2 Tungku Stainless Steel.',
        description: 'Kompor gas legendaris dari Rinnai yang terkenal awet dan api biru yang stabil. Mudah dibersihkan karena menggunakan bahan stainless steel berkualitas.',
        specs: {
            'Tipe': '2 Tungku',
            'Bahan': 'Stainless Steel',
            'Fitur': 'Api Ekonomis, Tatakan Enamel'
        }
    }

};

const categoryFallbacks = {
    'AC': {
        highlights: ['Pendinginan cepat & merata', 'Hemat listrik (Low Watt)', 'Operasi senyap'],
        sellingPoints: ['Garansi kompresor panjang', 'Tahan karat (Anti-Corrosion)', 'Layanan servis cepat'],
        objections: ['"Listrik jeglek?" -> Tidak, ini tipe Low Watt.', '"Pasang mahal?" -> Paket kami sudah termasuk pipa & pasang standar.']
    },
    'TV': {
        highlights: ['Warna tajam & realistis', 'Desain slim & modern', 'Konektivitas lengkap'],
        sellingPoints: ['Hemat energi', 'Panel kualitas grade A+', 'Garansi resmi terpercaya'],
        objections: ['"Cepat rusak panel?" -> Pakai stabilizer dan garansi kami cover panel.', '"Ribet setting?" -> Setting awal akan dibantu teknisi kami.']
    },
    'KULKAS': {
        highlights: ['Menjaga kesegaran lebih lama', 'Rak tempered glass kuat', 'Hemat energi'],
        sellingPoints: ['Bunga es minim', 'Ruang penyimpanan luas', 'Desain eksterior mewah'],
        objections: ['"Boros?" -> Teknologi baru lebih hemat 30%.', '"Bau?" -> Dilengkapi filter penyaring bau (Deodorizer).']
    },
    'MESIN CUCI': {
        highlights: ['Bersih maksimal, lembut di kain', 'Hemat air & deterjen', 'Body anti karat & tikus'],
        sellingPoints: ['Pengoperasian sangat mudah', 'Motor tangguh & awet', 'Pakaian lebih cepat kering'],
        objections: ['"Baju rusak?" -> Pulsator didesain khusus agar tidak melilit.', '"Berisik?" -> Sangat stabil bahkan saat memeras (spin).']
    },
    'SEPEDA LISTRIK': {
        highlights: ['Bebas bensin & polusi', 'Biaya charge sangat murah', 'Mudah dikendarai'],
        sellingPoints: ['Bisa dicicil tanpa kartu kredit', 'Servis bisa panggil ke rumah', 'Baterai awet & tahan lama'],
        objections: ['"Takut kena hujan?" -> Komponen sudah waterproof standar IP.', '"Sparepart susah?" -> Kami sedia lengkap dari baterai sampai ban.']
    },
    'HP': {
        highlights: ['Kamera jernih & tajam', 'Performa kencang anti lag', 'Baterai awet seharian'],
        sellingPoints: ['Garansi resmi 1 tahun', 'Bisa trade-in (tukar tambah)', 'Bonus aksesoris lengkap'],
        objections: ['"Gampang panas?" -> Sudah pakai sistem pendingin terbaru.', '"Memori penuh?" -> Dukungan slot MicroSD hingga 1TB.']
    }
};

function enrichProduct(p) {
    const name = p.name.toUpperCase();
    let enriched = { ...p };

    // Find model match
    for (const [key, data] of Object.entries(modelData)) {
        if (name.includes(key)) {
            enriched = { ...enriched, ...data };
            break;
        }
    }

    // Apply category fallbacks for missing fields
    const fallback = categoryFallbacks[p.category] || {};
    if (!enriched.shortDesc) {
        enriched.shortDesc = `${p.category} ${p.name} berkualitas tinggi untuk kebutuhan Anda.`;
    }
    if (!enriched.description) {
        enriched.description = `${p.name} adalah produk pilihan di kategori ${p.category} yang menawarkan performa handal dan efisiensi terbaik. Produk ini dirancang untuk memberikan kenyamanan maksimal bagi pengguna.`;
    }
    if (!enriched.specs || Object.keys(enriched.specs).length === 0) {
        enriched.specs = {
            'Tipe': p.name,
            'Kategori': p.category,
            'Kualitas': 'Original Tridjaya'
        };
    }
    if (!enriched.highlights) enriched.highlights = fallback.highlights || ['Kualitas Terjamin', 'Garansi Resmi', 'Layanan Aftersales'];
    if (!enriched.sellingPoints) enriched.sellingPoints = fallback.sellingPoints || ['Harga Bersaing', 'Bisa Cicilan', 'Barang Ready'];
    if (!enriched.objections) enriched.objections = fallback.objections || ['"Bisa kurang?" -> Harga sudah terbaik dengan layanan prima.', '"Garansi?" -> Resmi pabrikan.'];

    return enriched;
}

// Read generate_seeds.js
const seedFile = 'c:/Users/acer/Desktop/Project/RUST/Tridjaya Samrat/backend/scripts/generate_seeds.js';
let content = fs.readFileSync(seedFile, 'utf8');

// Find the products array
const startMarker = 'const products = [';
const endMarker = '];';
const startIndex = content.indexOf(startMarker) + startMarker.length;
const endIndex = content.indexOf(endMarker, startIndex);

const productsRaw = content.substring(startIndex, endIndex);
// This is a bit tricky because it's JS code, not JSON. 
// But since it's a static array in the script, we can eval it in a sandbox or just parse it if it's clean.
// For safety, I'll use a simpler regex or manual string manipulation.

// Let's try to parse it as an array of objects
const products = eval('[' + productsRaw + ']');
const enrichedProducts = products.map(enrichProduct);

// Stringify back
const enrichedRaw = enrichedProducts.map(p => {
    return `    {
        id: ${JSON.stringify(p.id)},
        name: ${JSON.stringify(p.name)},
        category: ${JSON.stringify(p.category)},
        price: ${p.price},
        image: ${JSON.stringify(p.image)},
        shortDesc: ${JSON.stringify(p.shortDesc)},
        description: ${JSON.stringify(p.description)},
        specs: ${JSON.stringify(p.specs, null, 8).replace(/\n/g, '\n    ')},
        highlights: ${JSON.stringify(p.highlights)},
        sellingPoints: ${JSON.stringify(p.sellingPoints)},
        objections: ${JSON.stringify(p.objections)}
    }`;
}).join(',\n');

const newContent = content.substring(0, startIndex) + '\n' + enrichedRaw + '\n' + content.substring(endIndex);
fs.writeFileSync(seedFile, newContent);
console.log('Successfully enriched 179 products in generate_seeds.js');

File: backend/scripts/enrich_products.py
import sqlite3

# Mapping: Keyword in Product Name -> Data Dict
DATA = {
    # MESIN CUCI
    "WS80SK10": {
        "short_desc": "Mesin cuci 2 tabung (Twin Tub) kapasitas 8kg hemat listrik (Super Low Watt) dengan fitur Hijab Care.",
        "description": "Hisense WS80SK10 adalah solusi pencucian efisien untuk keluarga Indonesia. Dilengkapi teknologi Hijab Care untuk pencucian lembut kain sensitif, Air Tub Spin Drying untuk pengeringan lebih cepat dan anti-bau, serta bodi plastik anti-karat yang tahan lama.",
        "specs": "Tipe: Twin Tub (Top Loading); Kapasitas: 8.0 Kg; Daya Cuci: 360-400W; Daya Spin: 160-180W; Kecepatan Spin: 1350 RPM; Dimensi: 763 x 450 x 892 mm."
    },
    "QW-9031HT": {
        "short_desc": "Mesin cuci 2 tabung 9kg 'Hijab Series' dengan fitur pengeringan Air Dry dan Dynamic Soak.",
        "description": "Aqua QW-9031HT dirancang khusus untuk perawatan hijab dan pakaian halus. Memiliki fitur Fresh Air System untuk pengeringan optimal tanpa bau apek, serta perendaman dinamis (Dynamic Soak) untuk mempermudah pengangkatan kotoran membandel.",
        "specs": "Tipe: Twin Tub; Kapasitas: 9 Kg; Daya Cuci: 380-400W; Daya Spin: 160W; Dimensi: 760 x 441 x 939 mm; Material: Kabinet Plastik PP."
    },
    "ES-T1290WA": {
        "short_desc": "Mesin cuci 2 tabung kapasitas jumbo 12kg dengan teknologi Dolphinwave dan filter Aquamagic.",
        "description": "Sharp ES-T1290WA Dolphinwave Series menawarkan kapasitas besar untuk keluarga besar. Dilengkapi W-Screw Pulsator untuk pusaran air tornado yang kuat, Super Aquamagic Filter dengan Ion Ag+ anti-bakteri, serta bodi plastik yang bebas korosi.",
        "specs": "Tipe: Twin Tub; Kapasitas: 12 Kg (Cuci) / 10 Kg (Spin); Daya Cuci: 360W; Daya Spin: 180W; Dimensi: 921 x 1065 x 551 mm."
    },
    "P1200RT": {
        "short_desc": "Mesin cuci 2 tabung 12kg dengan teknologi Roller Jet Pulsator dan Wind Jet Dry.",
        "description": "LG P-1200RT menawarkan performa pencucian maksimal dengan teknologi Roller Jet Pulsator yang menambah gesekan untuk hasil lebih bersih. Dilengkapi Wind Jet Dry untuk sirkulasi udara optimal saat pengeringan dan 3 program pencucian (Gentle, Normal, Strong).",
        "specs": "Tipe: Twin Tub; Kapasitas: 12 Kg; Daya: 450W; Dimensi: 880 x 530 x 1025 mm; Fitur: Punch+3, Wind Jet Dry."
    },
    "P-1200RT": {
        "short_desc": "Mesin cuci 2 tabung 12kg dengan teknologi Roller Jet Pulsator dan Wind Jet Dry.",
        "description": "LG P-1200RT menawarkan performa pencucian maksimal dengan teknologi Roller Jet Pulsator yang menambah gesekan untuk hasil lebih bersih. Dilengkapi Wind Jet Dry untuk sirkulasi udara optimal saat pengeringan dan 3 program pencucian (Gentle, Normal, Strong).",
        "specs": "Tipe: Twin Tub; Kapasitas: 12 Kg; Daya: 450W; Dimensi: 880 x 530 x 1025 mm; Fitur: Punch+3, Wind Jet Dry."
    },
    # SEPEDA LISTRIK
    "GODA 118": {
        "short_desc": "Sepeda listrik modern 600W dengan kecepatan 40km/jam and fitur Smart Key.",
        "description": "Goda Apollo GD 118 dirancang untuk mobilitas perkotaan yang efisien. Memiliki 3 mode kecepatan, fitur Auto-P untuk keamanan parkir, serta IoT Start/Smart Key untuk kemudahan akses tanpa kunci konvensional.",
        "specs": "Motor: 600W; Baterai: 48V 12Ah; Kecepatan: +/- 40 km/jam; Jarak Tempuh: +/- 40 km; Ban: Tubeless; Rem: Tromol."
    },
    "GODA 122": {
        "short_desc": "Sepeda listrik stylish 550W dengan intelligent display dan colorful running lights.",
        "description": "Goda Halo 122 menawarkan desain futuristik dengan lampu hias warna-warni dan dashboard fungsional. Sangat ekonomis dan ramah lingkungan, cocok untuk aktivitas harian seperti ke sekolah atau pasar.",
        "specs": "Motor: 550W; Baterai: 48V 12Ah; Kecepatan: +/- 40 km/jam; Jarak Tempuh: 35-40 km; Beban Maks: 150 kg; Fitur: Keyless, Remote Alarm."
    },
    "U-WINFLY D65": {
        "short_desc": "Sepeda listrik praktis 600W dengan sistem Smart Key NFC dan desain sporty.",
        "description": "U-Winfly D65 mengombinasikan kenyamanan berkendara dengan teknologi canggih seperti Smart Key NFC. Memiliki suspensi ganda yang empuk dan daya angkut hingga 150kg, ideal untuk komuter jarak pendek.",
        "specs": "Motor: 600W; Baterai: 48V 12Ah; Kecepatan: 33-45 km/jam; Jarak Tempuh: 40-50 km; Fitur: Remote Alarm, LED Headlight."
    },
    "SAIGE LUNA": {
        "short_desc": "Sepeda listrik elegan 800W dengan daya tempuh jauh hingga 70km dan tampilan mewah.",
        "description": "Saige Luna tampil dengan desain premium menyerupai motor listrik minimalis. Dengan motor 800W dan baterai 20Ah, sepeda ini mampu menempuh jarak yang lebih jauh dibandingkan model standar, memberikan kenyamanan ekstra untuk perjalanan harian.",
        "specs": "Motor: 800W; Baterai: 48V 20Ah; Kecepatan: +/- 45 km/jam; Jarak Tempuh: 55-70 km; Beban Maks: 200 kg; Fitur: Lock Motor, Find Vehicle."
    },
    "GODA 150A": {
        "short_desc": "Sepeda listrik tangguh 550W dengan jarak tempuh 40km dan sistem pengereman tromol.",
        "description": "Goda 150A adalah kendaraan listrik yang andal untuk kebutuhan transportasi jarak pendek. Didesain dengan struktur kokoh dan fitur keamanan standar yang memadai untuk kenyamanan berkendara di komplek atau perumahan.",
        "specs": "Motor: 550W; Baterai: 48V 12Ah; Kecepatan: 40 km/jam; Jarak Tempuh: 40 km; Beban Maks: 150 kg."
    },
    # HANDPHONE
    "SAMSUNG GALAXY A56": {
        "short_desc": "Smartphone premium 5G dengan Exynos 1580, layar Super AMOLED 120Hz, dan pengisian cepat 45W.",
        "description": "Samsung Galaxy A56 5G menghadirkan performa kencang dengan chipset Exynos 1580 (4nm) dan layar 120Hz yang sangat cerah (1900 nits). Dilengkapi kamera utama 50MP OIS dan sertifikasi IP67 untuk ketahanan air dan debu.",
        "specs": "Layar: 6.7 inch Super AMOLED 120Hz; Chipset: Exynos 1580; RAM/Storage: 8GB/256GB; Kamera: 50MP OIS + 12MP + 5MP; Baterai: 5000mAh (45W Fast Charging)."
    },
    "SAMSUNG A56": {
        "short_desc": "Smartphone premium 5G dengan Exynos 1580, layar Super AMOLED 120Hz, dan pengisian cepat 45W.",
        "description": "Samsung Galaxy A56 5G menghadirkan performa kencang dengan chipset Exynos 1580 (4nm) dan layar 120Hz yang sangat cerah (1900 nits). Dilengkapi kamera utama 50MP OIS dan sertifikasi IP67 untuk ketahanan air dan debu.",
        "specs": "Layar: 6.7 inch Super AMOLED 120Hz; Chipset: Exynos 1580; RAM/Storage: 8GB/256GB; Kamera: 50MP OIS + 12MP + 5MP; Baterai: 5000mAh (45W Fast Charging)."
    },
    "HOT 40i": {
        "short_desc": "HP Entry-level performa gaming dengan RAM besar (8GB+8GB virtual) and penyimpanan 256GB.",
        "description": "Infinix Hot 40i dirancang untuk multitasking lancar dan hiburan. Memiliki layar 90Hz yang halus, fitur Magic Ring interaktif untuk notifikasi, serta penyimpanan internal luas 256GB yang jarang ada di kelasnya.",
        "specs": "Layar: 6.56 inch 90Hz HD+; Chipset: Unisoc T606; RAM/Storage: 8GB/256GB; Kamera: 32MP Front / 50MP AI Rear; Baterai: 5000mAh (18W)."
    },
    "OPPO A98 5G": {
        "short_desc": "Smartphone 5G dengan 67W SUPERVOOC, layar 120Hz, and kamera mikroskop 40x.",
        "description": "OPPO A98 5G menonjolkan kecepatan pengisian daya super cepat (0-100% dalam 44 menit) and fitur kamera mikroskop unik. Didesain dengan OPPO Glow yang cantik and Battery Health Engine untuk masa pakai baterai hingga 4 tahun.",
        "specs": "Layar: 6.72 inch 120Hz FHD+; Chipset: Snapdragon 695 5G; RAM/Storage: 8GB/256GB; Kamera: 64MP + 2MP + 2MP Microscope; Baterai: 5000mAh (67W)."
    },
    "REDMI 10C": {
        "short_desc": "HP efisien bertenaga Snapdragon 680 dengan kamera 50MP and layar luas 6.71 inci.",
        "description": "Redmi 10C adalah pilihan cerdas untuk penggunaan harian. Ditenagai prosesor Snapdragon 680 yang hemat daya namun bertenaga, didukung kamera 50MP yang jernih and penyimpanan UFS 2.2 yang cepat untuk membuka aplikasi.",
        "specs": "Layar: 6.71 inch HD+; Chipset: Snapdragon 680; RAM/Storage: 4GB/64-128GB; Kamera: 50MP + 2MP; Baterai: 5000mAh (18W)."
    },
    "REDMI 12C": {
        "short_desc": "Smartphone ekonomis dengan layar besar 6.71 inci and kamera 50MP AI.",
        "description": "Redmi 12C menawarkan performa harian yang handal dengan harga terjangkau. Memiliki layar lega untuk menonton video, baterai 5000mAh yang awet, serta kamera 50MP untuk menangkap momen penting dengan detail.",
        "specs": "Layar: 6.71 inch HD+; Prosesor: Helio G85; RAM/Storage: hingga 4GB/128GB; Kamera: 50MP AI; Baterai: 5000mAh."
    },
    # TV
    "LE50AQT7000QU": {
        "short_desc": "Smart Android TV 50 inci 4K UHD dengan teknologi HQLED and Hands-free Voice Control.",
        "description": "AQUA LE50AQT7000QU Pro menghadirkan kualitas gambar memukau dengan HQLED 4K HDR and Dolby Vision. Dilengkapi sistem Android 11, fitur Google Assistant tanpa remote (Hands-free), and IoT Hub untuk mengontrol perangkat pintar di rumah Anda.",
        "specs": "Layar: 50 inch 4K UHD; Sistem: Android 11; Fitur: Dolby Atmos, Chromecast, HQLED; Daya: 160W; Dimensi: 1116 x 252 x 727 mm."
    },
    "LE55AQT7000QU": {
        "short_desc": "Smart Android TV 55 inci 4K UHD desain Bezel Less dengan visual HQLED yang nyata.",
        "description": "AQUA LE55AQT7000QU menawarkan pengalaman menonton bioskop di rumah dengan layar 55 inci tanpa bingkai (Bezel Less). Didukung teknologi HQ-LED 4K HDR untuk detail gambar yang tajam and fitur cerdas Google Assistant untuk kemudahan navigasi.",
        "specs": "Layar: 55 inch 4K UHD; Sistem: Android OS; Fitur: Intelligent Voice Control, IoT Hub, Chromecast; Daya: 150W; Dimensi: 1239 x 313 x 777 mm."
    },
    "POLYTRON 32": {
        "short_desc": "Smart Google TV 32 inci HD Ready dengan garansi 5 tahun and kualitas suara unggul.",
        "description": "TV Polytron 32 inci (Seri Cinemax/Google TV) memberikan gambar jernih dengan DIPE Engine and dukungan siaran TV Digital (DVB-T2). Dikenal dengan fitur suara yang kuat and jaminan garansi 5 tahun yang mencakup panel LED and remote.",
        "specs": "Layar: 32 inch HD Ready; Sistem: Google TV / Android; Fitur: DIPE Engine, USB Multimedia, T2 Digital; Garansi: 5 Tahun."
    },
    # KULKAS
    "SJ-236MG-GB": {
        "short_desc": "Kulkas 2 pintu 205L desain Shine Magneglas yang elegan and hemat energi.",
        "description": "Sharp SJ-236MG-GB Shine Series hadir dengan pintu bermotif geometris yang mewah and tahan gores. Dilengkapi Fan Cooling System untuk pendinginan merata, fitur 7 Shields untuk perlindungan maksimal, and teknologi AG+ Nano Deodorizer untuk menjaga kesegaran tanpa bau.",
        "specs": "Kapasitas: 205 Liter; Sistem: Fan Cooling; Daya: 100-130W; Fitur: Tempered Glass Tray, Ice Twist, 7 Shields; Dimensi: 545 x 588 x 1380 mm."
    }
}

def main():
    conn = sqlite3.connect('tridjaya.db')
    c = conn.cursor()
    
    updated_count = 0
    for kw, vals in DATA.items():
        # Match by keyword in name
        query = "UPDATE products SET short_desc = ?, description = ?, specs = ? WHERE name LIKE ?"
        params = (vals['short_desc'], vals['description'], vals['specs'], f'%{kw}%')
        c.execute(query, params)
        updated_count += c.rowcount
        if c.rowcount > 0:
            print(f"Updated {c.rowcount} products matching keyword: {kw}")
        
    conn.commit()
    conn.close()
    print(f"Total products enriched: {updated_count}")

if __name__ == "__main__":
    main()
